openupgradelib/openupgrade.py

```diff
@@ -611,18 +611,34 @@ def rename_fields(env, field_spec, no_deep=False):
             WHERE name = %s
                 AND type = 'model'
             """, (
-                "%s,%s" % (model, old_field),
                 "%s,%s" % (model, new_field),
+                "%s,%s" % (model, old_field),
             ),
         )
         # Rename appearances on export profiles
-        # TODO: Rename when the field is part of a submodel (ex. m2one.field)
-        cr.execute("""
-            UPDATE ir_exports_line
-            SET name = %s
-            WHERE name = %s
-            """, (old_field, new_field),
-        )
+        if column_exists(cr, "ir_exports", "model_id"):
+            # from OCA/server-tools/base_export_manager
+            cr.execute("""
+                UPDATE ir_exports_line iel
+                SET name = %s
+                FROM ir_exports ie,
+                    ir_model im
+                WHERE iel.name = %s
+                    AND ie.id = iel.export_id
+                    AND im.id = ie.model_id
+                    AND im.model = %s
+                """, (new_field, old_field, model),
+            )
+        elif column_exists(cr, "ir_exports", "resource"):
+            cr.execute("""
```

Comment:
Why not using always this SQL then?

Reply:
As I see, base_export_manager makes required=False the resource, and let's people choose between using model or resource.

I think it's ok now.